server/app/services/ti/TrustedIntermediaryService.java

```diff
@@ -251,49 +255,103 @@ public Form<EditTiClientInfoForm> updateClientInfo(
-    Optional<String> filteredNameQuery =
-        searchParameters.nameQuery().filter(s -> !s.isEmpty() && !s.isBlank());
-    Optional<String> filteredDateQuery =
-        searchParameters.dateQuery().filter(s -> !s.isEmpty() && !s.isBlank());
-    if (filteredNameQuery.isEmpty() && filteredDateQuery.isEmpty()) {
+    List<SearchParameters.ParamTypes> missingParams = findMissingSearchParams(searchParameters);
+    if (missingParams.size() == 4) {
       return TrustedIntermediarySearchResult.success(allAccounts);
     }
     final ImmutableList<AccountModel> searchedResult;
     try {
-      searchedResult = searchAccounts(searchParameters, allAccounts);
+      searchedResult = searchAccounts(searchParameters, allAccounts, missingParams);
     } catch (DateTimeParseException e) {
-      return TrustedIntermediarySearchResult.fail(
-          allAccounts, "Please enter date in MM/dd/yyyy format");
+      return TrustedIntermediarySearchResult.fail(allAccounts, "Please enter a valid birth date.");
     }
     return TrustedIntermediarySearchResult.success(searchedResult);
   }
 
   private ImmutableList<AccountModel> searchAccounts(
-      SearchParameters searchParameters, ImmutableList<AccountModel> allAccounts) {
+      SearchParameters searchParameters,
+      ImmutableList<AccountModel> allAccounts,
+      List<SearchParameters.ParamTypes> missingParams) {
+    Optional<LocalDate> maybeDOB =
+        validateAndConvertSearchParamDOB(searchParameters, missingParams);
     return allAccounts.stream()
         .filter(
             account ->
                 ((account.newestApplicant().get().getApplicantData().getDateOfBirth().isPresent()
-                        && searchParameters.dateQuery().isPresent()
-                        && !searchParameters.dateQuery().get().isEmpty()
+                        && maybeDOB.isPresent()
                         && account
                             .newestApplicant()
                             .get()
                             .getApplicantData()
                             .getDateOfBirth()
                             .get()
-                            .equals(
-                                dateConverter.parseIso8601DateToLocalDate(
-                                    searchParameters.dateQuery().get())))
-                    || (searchParameters.nameQuery().isPresent()
-                        && !searchParameters.nameQuery().get().isEmpty()
+                            .equals(maybeDOB.get()))
+                    || (!missingParams.contains(SearchParameters.ParamTypes.NAME)
                         && account
                             .getApplicantName()
                             .toLowerCase(Locale.ROOT)
                             .contains(
                                 searchParameters.nameQuery().get().toLowerCase(Locale.ROOT)))))
         .collect(ImmutableList.toImmutableList());
   }
+
+  private static List<SearchParameters.ParamTypes> findMissingSearchParams(
```

Comment:
Please revert static.

Reply:
This is static because this function is called from `validateSearch`, which is static for the reason described below.